dotnet/src/Microsoft.Agents.AI.CosmosNoSql/CosmosChatMessageStore.cs

```diff
@@ -0,0 +1,666 @@
+        }
+
+        return 0;
+    }
+
+    /// <summary>
+    /// Deletes all messages in this conversation.
+    /// This is an additional utility method beyond the base contract.
+    /// </summary>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The number of messages deleted.</returns>
+    public async Task<int> ClearMessagesAsync(CancellationToken cancellationToken = default)
+    {
+#pragma warning disable CA1513 // Use ObjectDisposedException.ThrowIf - not available on all target frameworks
+        if (this._disposed)
+        {
+            throw new ObjectDisposedException(this.GetType().FullName);
+        }
+#pragma warning restore CA1513
+
+        // Batch delete for efficiency
+        var query = new QueryDefinition("SELECT VALUE c.id FROM c WHERE c.conversationId = @conversationId AND c.Type = @type")
+            .WithParameter("@conversationId", this.ConversationId)
+            .WithParameter("@type", "ChatMessage");
+
+        var iterator = this._container.GetItemQueryIterator<string>(query, requestOptions: new QueryRequestOptions
+        {
+            PartitionKey = new PartitionKey(this.ConversationId),
+            MaxItemCount = this.MaxItemCount
+        });
+
+        var deletedCount = 0;
+        var partitionKey = new PartitionKey(this.ConversationId);
+
+        while (iterator.HasMoreResults)
+        {
+            var response = await iterator.ReadNextAsync(cancellationToken).ConfigureAwait(false);
+            var batch = this._container.CreateTransactionalBatch(partitionKey);
+            var batchItemCount = 0;
+
+            foreach (var itemId in response)
+            {
+                if (!string.IsNullOrEmpty(itemId))
+                {
+                    batch.DeleteItem(itemId);
+                    batchItemCount++;
+                    deletedCount++;
+                }
+            }
+
+            if (batchItemCount > 0)
+            {
+                await batch.ExecuteAsync(cancellationToken).ConfigureAwait(false);
+            }
+        }
+
+        return deletedCount;
+    }
+
+    /// <inheritdoc />
+    public void Dispose()
+    {
+        if (!this._disposed)
+        {
+            if (this._ownsClient)
+            {
+                this._cosmosClient?.Dispose();
+            }
+            this._disposed = true;
+        }
+    }
+
+    private sealed class StoreState
+    {
+        public string ConversationIdentifier { get; set; } = string.Empty;
+        public string? TenantId { get; set; }
+        public string? UserId { get; set; }
+        public bool UseHierarchicalPartitioning { get; set; }
+    }
+
+    /// <summary>
+    /// Represents a document stored in Cosmos DB for chat messages.
+    /// </summary>
+    [SuppressMessage("Performance", "CA1812:Avoid uninstantiated internal classes", Justification = "Instantiated by Cosmos DB operations")]
+    private sealed class CosmosMessageDocument
```

Comment:
One thing I've done in the past in Semantic Kernel, was forcing the `CosmosClient` to use System.Text.Json. Of course, it means you have to  control the settings that go to the client.